inspirehep/modules/migrator/models.py

```diff
@@ -73,12 +74,20 @@ def from_marcxml(cls, raw_record):
 
         The record must have a ``001`` tag containing the recid, otherwise it raises a ValueError.
         """
+        marc_record = create_record(raw_record, keep_singletons=False)
         try:
-            recid = int(cls.re_recid.search(raw_record).group('recid'))
```

Comment:
FYI the reason it was done like this is to be faster, as the recid is all you need to extract from the marcxml, so it seemed silly to build the whole `GroupableOrderedDict`.

Reply:
@michamos do you think I should do it the same way for the collections as well?

Reply:
it's way harder to do this for collections, so I think it's better like you are doing now.

Reply:
Ok, then I will do it like this for collection and the old way for recid :). Thanks!